proctor-consumer/src/main/java/com/indeed/proctor/consumer/AbstractGroupsManager.java

```diff
@@ -60,19 +69,23 @@ protected ProctorResult determineBucketsInternal(final Identifiers identifiers,
      * I don't see any value in using this in an application; you probably should use
      * {@link #determineBucketsInternal(HttpServletRequest, HttpServletResponse, Identifiers, Map, boolean)}
      *
-     * @param identifiers  a {@link Map} of unique-ish {@link String}s describing the request in the context of different {@link TestType}s.For example,
-     *                     {@link TestType#USER} has a CTK associated, {@link TestType#EMAIL} is an email address, {@link TestType#PAGE} might be a url-encoded String
-     *                     containing the normalized relevant page parameters
-     * @param context      a {@link Map} containing variables describing the context in which the request is executing. These will be supplied to any rules that
-     *                     execute to determine test eligibility.
-     * @param forcedGroups a {@link Map} from a String test name to an Integer bucket value. For the specified test allocate the specified bucket (if valid) regardless
-     *                     of the standard logic
+     * @param identifiers        a {@link Map} of unique-ish {@link String}s describing the request in the context of different {@link TestType}s.
+     *                           For example, {@link TestType#USER} has a CTK associated, {@link TestType#EMAIL} is an email address,
+     *                           {@link TestType#PAGE} might be a url-encoded String containing the normalized relevant page parameters
+     * @param context            a {@link Map} containing variables describing the context in which the request is executing.
+     *                           These will be supplied to any rules that execute to determine test eligibility.
+     * @param forceGroupsOptions a options to specify forced groups which includes a {@link Map} from a String test name to an Integer bucket value.
+     *                           For the specified test allocate the specified bucket (if valid) regardless of the standard logic
      * @return a {@link ProctorResult} to describe buckets allocations of all tests.
      */
     @VisibleForTesting
-    protected ProctorResult determineBucketsInternal(final Identifiers identifiers, final Map<String, Object> context, final Map<String, Integer> forcedGroups) {
+    protected ProctorResult determineBucketsInternal(
+            final Identifiers identifiers,
+            final Map<String, Object> context,
+            final ForceGroupsOptions forceGroupsOptions
+    ) {
         final GroupsManagerInterceptor interceptor = interceptorSupplier.get();
-        interceptor.beforeDetermineGroups(identifiers, context, forcedGroups);
+        interceptor.beforeDetermineGroups(identifiers, context, forceGroupsOptions.getForceGroups());
```

Comment:
Later is fine, but I think we want to update `GroupsManagerInterceptor` interface so that it takes `ForceGroupsOptions` as an argument instead of `forcedGroups`.

Reply:
Sure, Updated.

Reply:
I found it's not simple to update here. I will file a ticket.